background/get-addon-settings.js

```diff
@@ -4,10 +4,40 @@ import minifySettings from "../libraries/common/minify-settings.js";
  Since presets can change independently of others, we have to keep track of
  the versions separately. Current versions:
 
- - editor-dark-mode 2 (bumped in v1.23 twice)
- - editor-theme3 4 (last bumped in v1.29)
+ - editor-dark-mode 2 (bumped in v1.23 twice) (TODO: update this line)
+ - editor-theme3 3 (last bumped in v1.32)
  */
 
+const areColorsEqual = (currentColor, oldPresetColor) => {
+  // Case insensitive
+  const currentColorLowercase = currentColor.toLowerCase();
+  const oldPresetLowercase = oldPresetColor.toLowerCase();
+
+  // Converts  three/four/six value syntax into eight value syntax
+  const getRRGGBBAA = (hexColor) =>
+    hexColor.length === 7 // #{rr}{gg}{bb}  →  #{rr}{gg}{bb}ff
+      ? `${hexColor}ff`
+      : hexColor.length === 5 // #{r}{g}{b}{a}  →  #{rr}{gg}{bb}{aa}
+      ? `#${hexColor[1].repeat(2)}${hexColor[2].repeat(2)}${hexColor[3].repeat(2)}${hexColor[4].repeat(2)}`
+      : hexColor.length === 4 // #{r}{g}{b}  →  #{rr}{gg}{bb}ff
+      ? `#${hexColor[1].repeat(2)}${hexColor[2].repeat(2)}${hexColor[3].repeat(2)}ff`
+      : hexColor;
+
+  // Convert both colors to #{rr}{gg}{bb}{aa}
+  const currentColorRRGGBBAA = getRRGGBBAA(currentColorLowercase);
+  const oldPresetColorRRGGBBAA = getRRGGBBAA(oldPresetLowercase);
+
+  return currentColorRRGGBBAA === oldPresetColorRRGGBBAA;
+};
```

Comment:
@mxmou Let me know if I missed anything obvious when writing this function.
It rewrites all possible hex color formats to `#{rr}{gg}{bb}{aa}` format, so that the strings can be compared character-by-character and the strings will be equal if and only if both colors are equal.